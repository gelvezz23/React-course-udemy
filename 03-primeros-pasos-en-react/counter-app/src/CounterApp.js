/** @format */

import React from 'react';
import PropTypes from 'prop-types';

function CounterApp({ value }) {
	return (
		<>
			<h1>CounterApp</h1>
			<h2>{value}</h2>
		</>
	);
}

CounterApp.propTypes = {
	value: PropTypes.number,
};

export default CounterApp;
