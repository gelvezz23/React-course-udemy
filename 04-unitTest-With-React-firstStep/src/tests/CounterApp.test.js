/** @format */
import React from 'react';
import CounterApp from '../CounterApp';
import { shallow } from 'enzyme';

describe('Prueba a <CounterApp/>', () => {
	let wrapper = shallow(<CounterApp />);

	// en jest hay una especie de ciclo de vida llamado beforeEach()

	beforeEach(() => {
		// esto se ejecutara por cada prueba osea siempre que inicia una prueba
		// el se ejecuta con ella
		wrapper = shallow(<CounterApp />);
	});

	test('debe demostrar que <CounterApp/> renderice bien', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe mostrar el valor por defecto de 100', () => {
		const value = 100;
		const wrapper = shallow(<CounterApp value={value} />);
		const valor = wrapper.find('h2').text().trim();

		expect(valor).toBe('100');
	});

	test('Debe de incrementar al oprimir boton +1', () => {
		//.at(posicion en donde esta la etiqueta) es para buscar por posicion

		//.simulate('evento', {parametros}) sirve para simular un click en este boton
		const btn1 = wrapper.find('button').at(0).simulate('click');
		const valor = wrapper.find('h2').text().trim();

		expect(valor).toBe('24');
		// para leerlo es con .text o .html
		//console.log(btn1);
	});

	test('Debe de decrementar al oprimir boton -1', () => {
		//.at(posicion en donde esta la etiqueta) es para buscar por posicion

		//.simulate('evento', {parametros}) sirve para simular un click en este boton
		const btn2 = wrapper.find('button').at(2).simulate('click');
		console.log(btn2.text());
		const valor = wrapper.find('h2').text().trim();

		expect(valor).toBe('22');
		// para leerlo es con .text o .html
	});

	test('Debe de colocar el valor por defecto al dar click en btn reset', () => {
		const wrapper = shallow(<CounterApp value={100} />);
		wrapper.find('button').at(0).simulate('click');
		wrapper.find('button').at(0).simulate('click');
		wrapper.find('button').at(1).simulate('click');

		const valor = wrapper.find('h2').text().trim();
		expect(valor).toBe('100');
	});
});
