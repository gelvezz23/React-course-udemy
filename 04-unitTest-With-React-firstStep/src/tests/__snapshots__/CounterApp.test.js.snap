// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba a <CounterApp/> debe demostrar que <CounterApp/> renderice bien 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    23
  </h2>
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
    reset 
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
